### ISSUE TITLE
Scraping Moonlight games

### ISSUE THREAD

**Issue by CookiePLMonster** [muldjord/skyscraper#71]
I noticed that Skyscraper seems to be unable to scrape any entries from Moonlight games. Traditionally, launch .sh scripts for Moonlight integrated with ES (such as with https://github.com/CookiePLMonster/retromoonlight; my fork only fixes issues and doesn't change this specific functionality otherwise) are placed in "roms/moonlight" directory. Skyscraper detects this as a system correctly, albeit scraping seems to do nothing.

How does one attempt making this function as if this was a "pc" system, since that **should** make scraping work fine? I've attempted to add a "moonlight" (and "steam", too) alias to "pc" system with `Platform::getAliases`, albeit that didn't seem to make a difference.

**Comment by muldjord** [comment]
It's because the RetroPie scripts just lists all of your folders in roms (unrelated to whether Skyscraper knows them or not). And no, that won't work. You have to add the -i on command line or inputFolder in config.ini under [pc]

**Comment by muldjord** [comment]
If you don't currently have anything in the 'pc' folder you could also just create a symlink from the moonlight folder to the pc folder with this command:
```
$ ln -s /home/pi/RetroPie/roms/moonlight /home/pi/RetroPie/roms/pc
```
Then Skyscraper will just see the files from the moonlight folder inside the pc folder. Then it'll work with the RetroPie script aswell I believe. And then you don't need to add the -i or inputFolder configs.

**Comment by CookiePLMonster** [comment]
If platform aliases are a valid use case for this, would duplicating "pc" as new "moonlight" platform and then adding "pc" (and others from "pc" platform) as aliases allow it to be handled gracefully?

At that point I understand it's not fit for an upstream PR, but won't hurt to fork it for personal use and adapt.

Also somewhat unrelated question, excuse me if it's laid in plain sight in the docs and I overlooked it - is there a way to make Skyscraper ignore a specific file? RetroPie allows adding a shortcut to PIXEL as a "Desktop" rom in "ports", and Skyscraper picks that up.

I swear I glimpsed over a way to do it (special skyscraper.ignore files?) but I can't find it again now.

**Comment by muldjord** [comment]
EDIT: I misread. Yes, if you completely copy the pc platform to a new moonlight platform including the aliases that are there for the pc platform, it will work.

Second question: No, if it's a .sh file, Skyscraper will try to scrape it for the pc platform.

**Comment by CookiePLMonster** [comment]
So if aliases allow recognition at sources then a "moonlight" platform with "pc" alias would **potentially** work correctly. But as said, it's not something fit for an upstream PR and it may be easier said than done.

**Comment by muldjord** [comment]
Please reread, I just edited it. I misread your question.

**Comment by CookiePLMonster** [comment]
Fair enough!

Sadly I can't find any reference whether "moonlight" platform is an official convention for Moonlight or merely something thought of by retromoonlight maintainer. Without that I feel like adding it upstream would only bloat the code and potentially cause more questions (if Moonlight becomes a thing, what about Steam? etc. etc.). I'll handle this in a fork and manage Skyscraper manually as opposed to using Retropie-Setup.

Thanks for a timely response!

**Comment by muldjord** [comment]
My stance on this is that I only add platforms that are officially supported and described in the RetroPie readme's. So if you at some point manage to make that happen, I'll gladly add it. Until then I feel like it would just bloat the system. In general I am also a bit against these alias platforms that are basically just existing platforms using a different folder. There are quite a few of them already and I find them to be "ugly" so to speak. mame4all, fba and so on. They somewhat make sense, but it's not a platform as much as it is just a piece of software using roms from arcade (for instance). Same goes for moonlight. But let me know if you at some point get it on RetroPie officially.

**Comment by CookiePLMonster** [comment]
I just had a second look at `es_systems.cfg` because my current Moonlight entry caused incosistencies (it's named 'steam' but directory name is 'moonlight', so inbuilt ES scraper sees it as "STEAM" platform whereas skyscraper and Steven Selph's Scraper both see it as "moonlight") and realized something.

For example, take a look at "ports":
```
  <system>
    <name>ports</name>
    <fullname>Ports</fullname>
    <path>/home/pi/RetroPie/roms/ports</path>
    <extension>.sh .SH</extension>
    <command>bash %ROM%</command>
    <platform>pc</platform>
    <theme>ports</theme>
  </system>
```

Moonlight also has "pc" platform - therefore, wouldn't the issue be resolved reliably if RetroPie Setup Script queried that instead of directory name for roms? It would also potentially solve issues with "mame" systems, since there seem to be numerous systems for that with "arcade" platform.

That is, however, something to report in RetroPie Setup Script, I think.

EDIT:
Then again, I don't know how Skyscraper calls those internally, but the setup script explicitly calls them "systems"... That kinda explains my confusion, since what you refer to as "platforms" in the code is actually "systems", at least in EmulationStation.

It looks like things could be simplified if either
a) Setup Script operated on "platforms" instead of "systems" (may be tricky, since this may have some reasoning in the way ES GUI organizes games)
b) Skyscraper was aware of "platforms" vs "systems" confusion and translated it internally.

**Comment by CookiePLMonster** [comment]
**Just to resolve this matter and for future readers as well:**

I tried a `roms/pc` -> `roms/moonlight` symlink and that did not work with RetroPie Setup Script. The inverse, however (`roms/moonlight` -> `roms/pc`) works as expected - Skyscraper scrapes games fine, and EmulationStation shows them from `moonlight` directory.

EDIT:
I also had to symlink `.emulationstation/gamelists/moonlight` -> `.emulationstation/gamelists/pc`. When set up like this, both ES inbuilt scraper and Skyscraper work fine and can coexist.

Now to resolve an issue with MobyGames scraping, but that's a setup script posting a request for all games at once, breaching the 25 games-at-a-time MobyGames has =)
